### DISCUSSION TITLE
Inverse switch, cover, binary sensor

### DISCUSSION THREAD

**Discussion by disforw** [home-assistant/architecture#972] [Entity Models]
Looking to add native UI support to allow a user to simply invert any switch entity, binary sensor entity, or cover entity. This should be done using a small inverse checkmark inside the entity settings dialog box. Checking this setting can create an inverse helper (much like the Switch as X does) and hide the original entity.
This setting should be available for all switches, binary Sensors, fans and covers.
It doesn't seem that hard to make, I've already created a custom component that works really well for switches.

**Comment by farmio** [comment]
Same would be handy for percent values. 
Eg. to have cover states inverted to export it to other systems where 0% means open.

**Comment by frenck** [comment]
Going to close this one as a feature request, as not technical details for an architectural proposal have been provided. There is nothing to decide on.

../Frenck
